python/test/test_api.py

```diff
@@ -455,15 +455,20 @@ def test_basic_inference(self):
             "bool_input": numpy.random.rand(1, 100).astype(dtype=numpy.bool_),
         }
 
-        for response in server.model("test").infer(
+        response_iterator = server.model("test").infer(
             inputs=inputs,
             output_memory_type="cpu",
             raise_on_error=True,
-        ):
-            for input_name, input_value in inputs.items():
-                output_value = response.outputs[input_name.replace("input", "output")]
-                output_value = numpy.from_dlpack(output_value)
-                numpy.testing.assert_array_equal(input_value, output_value)
+        )
+
+        responses = list(response_iterator)
```

Comment:
This is subtle at a glance, but `list(response_iterator)` is actually blocking and iterating on the iterator until it's completely consumed, right?

Reply:
Correct. It looks like sometimes the previous version of the test didn't execute since there were no responses observed. This is to make sure that the response length is always one.